django/utils/deprecation.py

```diff
@@ -80,6 +80,10 @@ def __instancecheck__(self, instance):
 
 
 class MiddlewareMixin:
+
+    sync_capable = True
+    async_capable = False
+
```

Comment:
@claudep Is "no idea, I hoped a clear answer" would emerge sooner or later enough? Joking aside, I do not have good answers :(

Reply:
I wouldn't be shocked, but it really goes against "there should be one way of doing things". So all in all if anyone has ideas to finish what we started with the middleware move I'd happily use that over the status quo.

Is the code of your efforts online somewhere? Maybe we find a way to fix decorators :)

Reply:
Yes, I was sort of shocked when I found that basically all core middleware still uses `MiddlewareMixin`. I think there's still a path away from it, though - but maybe to more callable-style classes, rather than pure factory functions.